### PR TITLE
5.7.0

### PR DESCRIPTION
**Pull request by Renetnielsen**
## References

* #0000

### CODE REVIEW

File: azure/api-management/README.md [hunk 1]
```diff
@@ -30,23 +31,25 @@ This module creates the following resources.
...
 | `sku_name` | `string` | | **Required** | sku_name is a string consisting of two parts separated by an underscore(_). The first part is the name, valid values include: Consumption, Developer, Basic, Standard and Premium. The second part is the capacity (e.g. the number of deployed units of the sku), which must be a positive integer (e.g. Developer_1). |
+| `log_analytics_workspace_id` | `string` | | **Required** | ID of Log Analytics Workspace associated with the API Manager |
 | `tags` | `any` | `{}` | | A mapping of tags to assign to the resource. |
 
 ## Usage
 
 ```ruby
 module "appi_example" { 
-  source                = "git::https://github.com/Energinet-DataHub/geh-terraform-modules.git//azure/api-management?ref=5.1.0"
+  source                     = "git::https://github.com/Energinet-DataHub/geh-terraform-modules.git//azure/api-management?ref=5.6.0"
```
Comment (djorgensendk) [inline azure/api-management/README.md:41]:
```suggestion
  source                     = "git::https://github.com/Energinet-DataHub/geh-terraform-modules.git//azure/api-management?ref=5.7.0"
```

File: azure/api-management/README.md [hunk 2]
```diff
@@ -55,7 +58,7 @@ Two tags is added by default
 ```ruby
 locals {
   module_tags = {
-    "ModuleVersion" = "5.1.0"
+    "ModuleVersion" = "5.6.0"
```
Comment (djorgensendk) [inline azure/api-management/README.md:61]:
```suggestion
    "ModuleVersion" = "5.7.0"
```

File: azure/mssql-database/README.md [hunk 3]
```diff
@@ -35,14 +37,15 @@ This module creates the following resource:
 
 ```ruby
 module "sqldb_example" { 
-  source                        = "git::https://github.com/Energinet-DataHub/geh-terraform-modules.git//azure/sql-database?ref=5.4.0"
+  source                        = "git::https://github.com/Energinet-DataHub/geh-terraform-modules.git//azure/sql-database?ref=5.6.0"
```
Comment (djorgensendk) [inline azure/mssql-database/README.md:40]:
```suggestion
  source                        = "git::https://github.com/Energinet-DataHub/geh-terraform-modules.git//azure/sql-database?ref=5.7.0"
```

File: azure/mssql-database/README.md [hunk 4]
```diff
@@ -53,7 +56,7 @@ Two tags is added by default
 ```ruby
 locals {
   module_tags = {
-    "ModuleVersion" = "5.4.0"
+    "ModuleVersion" = "5.6.0"
```
Comment (djorgensendk) [inline azure/mssql-database/README.md:59]:
```suggestion
    "ModuleVersion" = "5.7.0"
```

File: azure/mssql-database/main.tf [hunk 5]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 locals {
   module_tags = {
-    "ModuleVersion" = "5.4.0"
+    "ModuleVersion" = "5.6.0"
```
Comment (djorgensendk) [inline azure/mssql-database/main.tf:16]:
```suggestion
    "ModuleVersion" = "5.7.0"
```

File: azure/mssql-server/README.md [hunk 6]
```diff
@@ -67,7 +70,7 @@ Two tags is added by default
 ```ruby
 locals {
   module_tags = {
-    "ModuleVersion" = "5.4.0"
+    "ModuleVersion" = "5.6.0"
```
Comment (djorgensendk) [inline azure/mssql-server/README.md:73]:
```suggestion
    "ModuleVersion" = "5.7.0"
```

File: azure/mssql-server/main.tf [hunk 7]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 locals {
   module_tags = {
-    "ModuleVersion" = "5.4.0"
+    "ModuleVersion" = "5.6.0"
```
Comment (djorgensendk) [inline azure/mssql-server/main.tf:16]:
```suggestion
    "ModuleVersion" = "5.7.0"
```

File: azure/service-bus-namespace/README.md [hunk 8]
```diff
@@ -45,7 +48,7 @@ This module creates the following resources.
 
 ```ruby
 module "service_bus_namespace_example" { 
-  source                = "git::https://github.com/Energinet-DataHub/geh-terraform-modules.git//azure/service_bus-namespace?ref=5.1.0"
+  source                = "git::https://github.com/Energinet-DataHub/geh-terraform-modules.git//azure/service_bus-namespace?ref=5.6.0"
```
Comment (djorgensendk) [inline azure/service-bus-namespace/README.md:51]:
```suggestion
  source                = "git::https://github.com/Energinet-DataHub/geh-terraform-modules.git//azure/service_bus-namespace?ref=5.7.0"
```

File: azure/service-bus-namespace/README.md [hunk 9]
```diff
@@ -74,7 +77,7 @@ Two tags is added by default
 ```ruby
 locals {
   module_tags = {
-    "ModuleVersion" = "5.1.0"
+    "ModuleVersion" = "5.6.0"
```
Comment (djorgensendk) [inline azure/service-bus-namespace/README.md:80]:
```suggestion
    "ModuleVersion" = "5.7.0"
```

**Review by djorgensendk — APPROVED** [review]

File: azure/mssql-database/variables.tf [hunk 10]
```diff
@@ -54,8 +54,19 @@ variable auto_pause_delay_in_minutes {
   description = "(Optional) Time in minutes after which database is automatically paused. A value of -1 means that automatic pause is disabled. This property is only settable for General Purpose Serverless databases."
 }
 
+variable log_analytics_workspace_id {
+  type = string
+  description = "(Required) The id of the Log Analytics Workspace where the SQL DB will log events (e.g. audit events)"
+}
+
+variable log_retention_in_days {
```
Comment (dstsyst) [inline azure/mssql-database/variables.tf:62]: Should be documented in Readme.md under "arguments" and under "usage"

File: azure/mssql-server/variables.tf [hunk 11]
```diff
@@ -66,6 +66,17 @@ variable firewall_rules {
   default     = []
 }
 
+variable log_analytics_workspace_id {
+  type = string
+  description = "(Required) The id of the Log Analytics Workspace where the SQL DB will log events (e.g. audit events)"
+}
+
+variable log_retention_in_days {
```
Comment (dstsyst) [inline azure/mssql-server/variables.tf:74]: Should be documented in Readme.md under "arguments" and under "usage".

File: azure/service-bus-namespace/main.tf [hunk 12]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 locals {
   module_tags = {
-    "ModuleVersion" = "5.1.0"
+    "ModuleVersion" = "5.6.0"
```
Comment (dstsyst) [inline azure/service-bus-namespace/main.tf:16]:
```suggestion
    "ModuleVersion" = "5.7.0"
```

File: azure/service-bus-namespace/variables.tf [hunk 13]
```diff
@@ -62,8 +62,20 @@ variable auth_rules {
   description = "(Required) A list of objects describing the Service Bus Namespace auth rules."
 }
 
+variable log_analytics_workspace_id {
+  type = string
+  description = "(Optional) The id of the Log Analytics Workspace where theService Bus will log events (e.g. audit events)"
+  default = null
+}
+
+variable log_retention_in_days {
```
Comment (dstsyst) [inline azure/service-bus-namespace/variables.tf:71]: Should be documented in Readme.md under "arguments" and "usage".

**Review by dstsyst — COMMENTED** [review]
Added a few comments

File: azure/mssql-server/variables.tf (hunk 11 above)
Comment (Renetnielsen) [reply azure/mssql-server/variables.tf:74]: Default variable, so its documented per say

File: azure/service-bus-namespace/variables.tf (hunk 13 above)
Comment (Renetnielsen) [reply azure/service-bus-namespace/variables.tf:71]: Default variable, so its documented per say

File: azure/mssql-database/variables.tf (hunk 10 above)
Comment (Renetnielsen) [reply azure/mssql-database/variables.tf:62]: Default variable, so its documented per say

File: azure/key-vault/variables.tf [hunk 14]
```diff
@@ -65,13 +65,19 @@ variable enabled_for_template_deployment {
   default     = false
 }
 
+variable log_analytics_workspace_id {
+  type        = string
+  description = "(Required) The id of the Log Analytics Workspace where the keyvault will log events (e.g. audit events)"
+}
+
+variable log_retention_in_days {
```
[the remark this answers is not in the document]
Comment (Renetnielsen) [reply azure/key-vault/variables.tf:73]: Default variable, so its documented per say

File: azure/api-management/variables.tf [hunk 15]
```diff
@@ -56,6 +56,11 @@ variable sku_name {
   description = "(Required) sku_name is a string consisting of two parts separated by an underscore(_). The first part is the name, valid values include: Consumption, Developer, Basic, Standard and Premium. The second part is the capacity (e.g. the number of deployed units of the sku), which must be a positive integer (e.g. Developer_1)."
 }
 
+variable log_analytics_workspace_id {
```
[the remark this answers is not in the document]
Comment (Renetnielsen) [reply azure/api-management/variables.tf:59]: optional variable, so its documented per say

File: azure/api-management/main.tf [hunk 16]
```diff
@@ -35,4 +35,19 @@ resource "azurerm_api_management" "this" {
...
+resource "azurerm_monitor_diagnostic_setting" "this" {
+  name                       = "diag-apim-${lower(var.name)}-${lower(var.project_name)}-${lower(var.environment_short)}-${lower(var.environment_instance)}"
+  target_resource_id         = azurerm_api_management.this.id
+  log_analytics_workspace_id = var.log_analytics_workspace_id
+
+  metric {
+    category = "AllMetrics"
+    enabled  = true
+
+    retention_policy {
```
[the remark this answers is not in the document]
Comment (Renetnielsen) [reply azure/api-management/main.tf:49]: added

**Review by dstsyst — APPROVED** [review]

**Pull request merged**